### PR TITLE
AK-13281 Update docs for cloud connector and provider

### PR DESCRIPTION
**Pull request by spikewang**
Update docs for cloud connector/provider and add template

+ Add the template for generate top level doc
+ Update docs for cloud connector (AWS, Azure, GCP)
+ Add examples
+ Update docs for segment and group

### CODE REVIEW

File: alkira/resource_alkira_connector_azure_vnet.go [hunk 1]
```diff
@@ -10,58 +10,63 @@ import (
...
+				Description: "The name of the connector.",
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "The name of the connector",
 			},
-			"segment": {
-				Type:     schema.TypeString,
-				Required: true,
+			"segment":       {
+				Description: "The size of the connector, one of `SMALL`, `MEDIUM` or `LARGE`.",
```
Comment (rvichery) [inline alkira/resource_alkira_connector_azure_vnet.go:62]: This line need to be switched with L67

File: alkira/resource_alkira_connector_azure_vnet.go [hunk 2]
```diff
@@ -10,58 +10,63 @@ import (
...
-			"azure_region": &schema.Schema{
+			"azure_region":  {
+				Description: "Region of Azure VNET.",
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "Azure Region",
 			},
-			"azure_vnet_id": &schema.Schema{
+			"azure_vnet_id": {
+				Description: "Azure Virutal Network Id.",
```
Comment (rvichery) [inline alkira/resource_alkira_connector_azure_vnet.go:27]:
```suggestion
				Description: "Azure Virtual Network Id.",
```

File: alkira/resource_alkira_connector_azure_vnet.go [hunk 3]
```diff
@@ -10,58 +10,63 @@ import (
...
+
 		Create: resourceConnectorAzureVnetCreate,
 		Read:   resourceConnectorAzureVnetRead,
 		Update: resourceConnectorAzureVnetUpdate,
 		Delete: resourceConnectorAzureVnetDelete,
 
 		Schema: map[string]*schema.Schema{
-			"azure_region": &schema.Schema{
+			"azure_region":  {
+				Description: "Region of Azure VNET.",
```
Comment (rvichery) [inline alkira/resource_alkira_connector_azure_vnet.go:22]: To be consistent with AWS
```suggestion
				Description: "Azure Region where VNET resides.",
```

File: alkira/resource_alkira_connector_azure_vnet.go [hunk 4]
```diff
@@ -10,58 +10,63 @@ import (
...
-			"credential_id": &schema.Schema{
+			"credential_id": {
+				Description: "The Credential to access Azure.",
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "The credentials for creating connector",
 			},
-			"cxp": &schema.Schema{
+			"cxp":           {
+				Description: "The CXP to be used for the connector.",
```
Comment (rvichery) [inline alkira/resource_alkira_connector_azure_vnet.go:47]:
```suggestion
				Description: "The CXP where the connector should be provisioned.",
```

File: alkira/resource_alkira_connector_azure_vnet.go [hunk 5]
```diff
@@ -10,58 +10,63 @@ import (
...
-			"connector_id": &schema.Schema{
-				Type:     schema.TypeInt,
-				Computed: true,
+			"connector_id":  {
+				Type:        schema.TypeInt,
+				Computed:    true,
 			},
-			"credential_id": &schema.Schema{
+			"credential_id": {
+				Description: "The Credential to access Azure.",
```
Comment (rvichery) [inline alkira/resource_alkira_connector_azure_vnet.go:42]:
```suggestion
				Description: "ID of credential managed by Credential Manager.",
```

File: alkira/resource_alkira_connector_azure_vnet.go [hunk 6]
```diff
@@ -10,58 +10,63 @@ import (
...
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "Azure Virutal Network Id",
 			},
-			"billing_tags": {
-				Type:     schema.TypeList,
-				Optional: true,
-				Elem:     &schema.Schema{Type: schema.TypeInt},
+			"billing_tags":  {
+				Description: "Tag for billing.",
```
Comment (rvichery) [inline alkira/resource_alkira_connector_azure_vnet.go:32]:
```suggestion
				Description: "Tags for billing.",
```

File: alkira/resource_alkira_connector_azure_vnet.go [hunk 7]
```diff
@@ -10,58 +10,63 @@ import (
...
-			"cxp": &schema.Schema{
+			"cxp":           {
+				Description: "The CXP to be used for the connector.",
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "The CXP to be used for the connector",
 			},
-			"group": &schema.Schema{
+			"group":         {
+				Description: "The group that the connector belongs to.",
```
Comment (rvichery) [inline alkira/resource_alkira_connector_azure_vnet.go:52]:
```suggestion
				Description: "The group of the connector.",
```

**Review by rvichery — COMMENTED** [review]
Added couple of comments to ensure consistency between different cloud connectors

File: alkira/resource_alkira_connector_azure_vnet.go (hunk 1 above)
Comment (spikewang) [reply alkira/resource_alkira_connector_azure_vnet.go:62]: opps

**Review by rvichery — APPROVED** [review]
LGTM. thanks for updating the descriptions.

**Pull request merged**